tree-sitter-stack-graphs/src/cli/init.rs

```diff
@@ -23,9 +23,10 @@ use std::path::PathBuf;
 const TSSG_VERSION: &str = env!("CARGO_PKG_VERSION");
 
 lazy_static! {
-    static ref VALID_NAME: Regex = Regex::new(r"^[a-zA-Z0-9_.-]+$").unwrap();
-    static ref VALID_NPM_PKG: Regex = Regex::new(r"^(@[a-zA-Z0-9_.-]+/)?[a-zA-Z0-9_.-]+$").unwrap();
-    static ref VALID_NPM_VERSION: Regex = Regex::new(r"^[0-9]+\.[0-9]+\.[0-9]+$").unwrap();
+    static ref VALID_CRATE_NAME: Regex = Regex::new(r"^[a-zA-Z_-][a-zA-Z0-9_-]*$").unwrap();
+    static ref VALID_CRATE_VERSION: Regex = Regex::new(r"^[0-9]+\.[0-9]+\.[0-9]+$").unwrap();
+    static ref VALID_DEPENDENCY_VERSION: Regex =
+        Regex::new(r"^[~^]?[0-9]+(\.[0-9]+(\.[0-9]+)?)?$").unwrap();
```

Comment:
Based on some feedback in `github/symbol-extraction`,  using `[once_cell](https://docs.rs/once_cell/1.17.0/once_cell/#lazy-initialized-global-data)` has the advantage of not relying on a macro rule and can be formatted and checked with the Rust tooling.  Would this be the right time to switch over to `once_cell` for these regex patterns?

I think it would mean these become:

```rust
use once_cell::sync::Lazy;
static VALID_CRATE_NAME: Lazy<Regex> =
    Lazy::new(|| Regex::new(r"^[a-zA-Z_-][a-zA-Z0-9_-]*$").unwrap());
...
```

Reply:
Addressed in 875defc.